src/main/java/io/cryostat/discovery/Discovery.java

```diff
@@ -339,7 +339,7 @@ public void publish(
                                             + " current URI range settings",
                                     b.target.connectUrl));
                 }
-                if (agentTlsRequired && !b.target.connectUrl.getScheme().equals("https")) {
+                if (agentTlsRequired && !plugin.callback.getScheme().equals("https")) {
```

Comment:
I think this should still be `b.target.connectUrl`, since the context here is that this is within a loop iterating over the published Target definitions. The URI range check immediately preceding this checks the target connection URL, too.

All plugin callback URI validations can/should be done when the plugin attempts to register itself. When the plugin has registered and is later attempting to publish discovery information into the tree, then we should validate the content it is publishing. At the publication step we could also re-validate the plugin itself (callback URL), but that's probably not really necessary.